From d3859630aa35969199fabcb084bd97719041158b Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date: Fri, 26 Oct 2018 15:14:16 +0100
Subject: [PATCH 277/773] staging: vchiq_arm: Register bcm2835-codec as a
 platform driver

Following the same pattern as bcm2835-camera and bcm2835-audio,
register the V4L2 codec driver as a platform driver

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
---
 drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c
+++ b/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c
@@ -171,6 +171,7 @@ static struct device *vchiq_dev;
 static DEFINE_SPINLOCK(msg_queue_spinlock);
 static struct platform_device *bcm2835_camera;
 static struct platform_device *bcm2835_audio;
+static struct platform_device *bcm2835_codec;
 
 static const char *const ioctl_names[] = {
 	"CONNECT",
@@ -3660,6 +3661,9 @@ static int vchiq_probe(struct platform_d
 	bcm2835_audio = vchiq_register_child(pdev, "bcm2835_audio");
 	if (IS_ERR(bcm2835_audio))
 		bcm2835_audio = NULL;
+	bcm2835_codec = vchiq_register_child(pdev, "bcm2835-codec");
+	if (IS_ERR(bcm2835_codec))
+		bcm2835_codec = NULL;
 
 	return 0;
 
